crates/ide/src/inlay_hints/implicit_drop.rs

```diff
@@ -54,7 +54,7 @@ pub(super) fn hints(
             };
             let range = match terminator.span {
                 MirSpan::ExprId(e) => match source_map.expr_syntax(e) {
-                    Ok(s) => {
+                    Ok(s) if s.file_id == file_id => {
```

Comment:
```suggestion
                    Ok(s) if !s.file_id.is_macro() => {
```
Just a minor note: can you add a comment that we're ignoring macros, or alternatively test for `!s.file_id.is_macro()` which is more self-documenting?